test/iterators.jl

```diff
@@ -22,6 +22,22 @@ let z = zip(1:2, 3:4, 5:6)
     @test eltype(z) == Tuple{Int,Int,Int}
 end
 
+# unzip
+let z = zip(1:2)
+    @test unzip(z) == 1:2
+    @test eltype(unzip(z)) == UnitRange{Int64}
```

Comment:
`UnitRange{Int}` here since it will fail on systems with `Int == Int32` (and same on the tests below)